app.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import logging
+import os
+import typing
+
+import boto3
+from django.db import transaction  # type: ignore
+from openstates.cli.reports import generate_session_report
+
+
+logger = logging.getLogger("openstates")
+s3_client = boto3.client('s3')
+
+
+def process_upload_function(event, context):
+    """
+    Process a file upload.
+    """
+
+    # Get the uploaded file's information
+    bucket = event['Records'][0]['s3']['bucket']['name']  # Will be `my-bucket`
+    key = event['Records'][0]['s3']['object']['key']  # Will be the file path of whatever file was uploaded.
+
+    # Get the bytes from S3
+    datadir = "/tmp/"
+
+    state_acronym = key.split("/")[0]  # e.g az, al, etc
+
+    name_of_file = key.split("/")[-1]  # e.g. bills.json, votes.json, etc
+    filedir = f"{datadir}{name_of_file}"
+
+    s3_client.download_file(bucket, key, filedir)  # Download this file to writable tmp space.
+
+    jurisdiction_id = f"ocd-jurisdiction/country:us/state:{state_acronym}/government"
```

Comment:
Probably better, again, to get this from request metadata if possible. Makes it more future-proof and easier to handle format changes.

Reply:
Got it @johnseekins - the metadata retrieved here is controlled by lambda itself, we're not sending anything in with the data uploaded. [I'll confirm the possibility of adding more details the event metadata]